libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Extensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    public static class Extensions
+    {
+        public static bool IsPropertyValue(this JToken value, string name) =>
+            value.Parent is JProperty jprop && jprop.Name == name;
+
+        public static T Dequeue<T>(this List<T> queue)
```

Comment:
>List [](start = 40, length = 4)

is there a reason you are using List<> instead of Queue<> ?

Reply:
Because I'm not sure how queues serialize/deserialize.  I'll look into it.

---
In reply to: [360172277](https://github.com/microsoft/botbuilder-dotnet/pull/2936#discussion_r360172277) [](ancestors = 360172277)